delta/sdk-testkit/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/OrganizationsDummy.scala

```diff
@@ -72,109 +50,68 @@ final class OrganizationsDummy private (
-    uuidToLabel.get.flatMap(_.get(uuid) match {
-      case Some(label) => fetchAt(label, rev)
+    fetch(uuid).flatMap {
+      case Some(value) => fetchAt(value.id, rev)
       case None        => IO.pure(None)
-    })
+    }
 
   override def list(
       pagination: Pagination.FromPagination,
       params: OrganizationSearchParams
   ): UIO[SearchResults.UnscoredSearchResults[OrganizationResource]] =
-    cache.get.map { resources =>
-      val filtered = resources.values.filter(params.matches).toVector.sortBy(_.createdAt)
-      UnscoredSearchResults(
-        filtered.length.toLong,
-        filtered.map(UnscoredResultEntry(_)).slice(pagination.from, pagination.from + pagination.size)
-      )
-    }
+    cache.list(pagination, params)
 
   override def events(offset: Offset = NoOffset): Stream[Task, Envelope[OrganizationEvent]] =
-    DummyHelpers.eventsFromJournal(journal.get, offset, maxStreamSize)
+    journal.events(offset)
 
   override def currentEvents(offset: Offset): Stream[Task, Envelope[OrganizationEvent]] =
-    DummyHelpers.currentEventsFromJournal(journal.get, offset, maxStreamSize)
-
-  private def setToCache(resource: OrganizationResource): UIO[Unit]              =
-    cache.update(_ + (resource.value.label -> resource))
-
-  private def setUuid(resource: OrganizationResource): UIO[OrganizationResource] =
-    uuidToLabel.update(_ + (resource.value.uuid -> resource.value.label)).as(resource)
-
-  private def currentState(label: Label): UIO[Option[OrganizationState]]         =
-    journalMap.map { labelsEvents =>
-      labelsEvents
-        .get(label)
-        .map(_.foldLeft[OrganizationState](Initial) { (s, e) => Organizations.next(s, e.event) })
-    }
-
-  private def stateAt(label: Label, rev: Long): IO[RevisionNotFound, Option[OrganizationState]] =
-    journalMap.flatMap { labelsEvents =>
-      labelsEvents.get(label).traverse { events =>
-        if (events.size < rev)
-          IO.raiseError(RevisionNotFound(rev, events.size.toLong))
-        else
-          events
-            .foldLeft[OrganizationState](Initial) {
-              case (state, envelope) if envelope.event.rev <= rev => Organizations.next(state, envelope.event)
-              case (state, _)                                     => state
-            }
-            .pure[UIO]
-      }
-    }
+    journal.currentEvents(offset)
 
   private def eval(cmd: OrganizationCommand): IO[OrganizationRejection, OrganizationResource] =
-    semaphore.withPermit {
-      for {
-        tgEvents <- journal.get
-        state    <- currentState(cmd.label).map(_.getOrElse(Initial))
-        event    <- Organizations.evaluate(state, cmd)
-        _        <- journal.set(tgEvents :+ makeEnvelope(event, s"$entityType-${event.label}", offsetMax))
-        res      <- IO.fromEither(Organizations.next(state, event).toResource.toRight(UnexpectedInitialState(cmd.label)))
-        _        <- setToCache(res)
-      } yield res
-    }
+    for {
```

Comment:
this is no longer safe without a semaphore when executed concurrently

Reply:
The semaphore is in the journal implementation